packages/web/src/components/spa/HackAUBG/RegistrationForm/RegistrationForm.jsx

```diff
@@ -0,0 +1,756 @@
+            .then((res) => {
+                setLoadingAnimation(false);
+                setApiError(false);
+            })
+            .catch((err) => {
+                setLoadingAnimation(false);
+                setErrorMessage(err);
+                setApiError(true); // put button in error state
+                setSubmitButtonValue('Retry');
+            });
+    };
+
+    function parseVars(data) {
+        //getting all data from form and converting True/False string to boolean
+        Object.entries(data).forEach((field) => {
+            if (field[1] === 'True' || field[1] === 'False') {
+                data[field[0]] = field[1] === 'True' ? true : false;
+            }
+        });
+        return data;
+    }
+
+    //checking what error code we receive from the backend and outputing message depending on the code
+    function setErrorMessage(err) {
+        if (err['response']['status'] >= 500) {
+            setButtonMessage('Something went wrong'); //add new message
+        } else {
+            setButtonMessage(err['response']['data']['message']);
+        }
+    }
+
+    //send data to registerMember function
+    const onSubmit = (data) => {
+        setLoadingAnimation(true);
+        data = parseVars(data);
+        data = checkTeamname(data);
+        registerMember(data);
+    };
+
+    // useEffect will always initialize this with the correct state
+    const [buttonState, setButtonState] = useState(
+        'hackaubg-register-btn disabled'
+    );
+
+    //handleDisabledFields will blur teamname inout field if they answer no
+    const handleDisabledFields = (e) => {
+        if (e.target.name === 'hasteam') {
+            if (e.target.value === 'False') {
+                setDisableTeamNameField(true);
+            } else {
+                setDisableTeamNameField(false);
+            }
+        }
+    };
+    //checks if user selected Yes/No on question about if they have team and if they do not have the teamname value is null
+    function checkTeamname(data) {
+        if (data.hasteam == false) {
+            data.teamname = null;
+        }
+        return data;
+    }
+
+    //changes the button color and text depending on what is enetred in the form
+    const checkButtonAvailability = () => {
+        if (apiError == true) {
+            setButtonState('hackaubg-register-btn error');
+            return;
+        } else if (apiError == false) {
+            if (Object.keys(errors).length != 0) {
+                setButtonState('hackaubg-register-btn error');
+                setSubmitPressed(false);
+                setSubmitButtonValue('Retry');
+                return;
+            } else if (Object.keys(errors).length == 0 && submitPressed) {
+                setButtonState('hackaubg-register-btn disabled');
+                setSubmitButtonValue('Success');
+                return;
+            }
+        }
+        setButtonState('hackaubg-register-btn ');
```

Comment:
is the className working as expected with the space at the end?

Reply:
it is working with both with and without it so i will remove it